### PR TITLE
Refactor timeseries management, time storage in database, and device settings

### PR DESCRIPTION
**Pull request by jhubaum**
More detail on the changes is in each commit message. Since I moved quite some code around, it's probably best to review this MR commit by commit

## Pre-merge checklist
- [x] Ensure that the deployment is still working

### CODE REVIEW

File: src/util/redis.py [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+        redis_key: The key under which the series will be stored
+        duplicate_policy: The redis policy to use if the series already exists in redis
+    """
+    pipe = redis.ts().pipeline()
+    for index, value in series.items():
+        pipe.add(
+            redis_key,
+            _datetime_to_redis_timestamp(index),
+            float(value),
+            labels=dict(value=series.name),
```
Comment (jhubaum) [inline src/util/redis.py:71]: I don't know what the `labels` parameter is used for and the redis documentation is, let's say... lacking. So I'm unsure if I'm setting it to the right value here. @philippschmalen Do you remember why you were setting this parameter in your implementation of this function?

Comment (philippschmalen) [reply src/util/redis.py:71]: It adds metadata to a time series. We could remove it as it does not change anything. I'll keep it for now to start testing. 

Official docs: 
"LABELS {label value}...
is set of label-value pairs that represent metadata labels of the time series.

Use it only if you are creating a new time series. It is ignored if you are adding samples to an existing time series. See LABELS in [TS.CREATE](https://redis.io/commands/ts.create)."

**Review by philippschmalen — APPROVED** [review]
Huge chunk of work, great stuff and hard to digest at once. So, let's move to testing to see how it works out and where we need to improve to achieve a MVP.

**Comment by jhubaum** [comment]
Just fyi, you should've pushed `merge` instead of `close` so that the changes actually land on master. No big deal though, I'll do that now

**Comment by philippschmalen** [comment]
My bad! I have no clue what I was doing

**Pull request merged**